XenCert/StorageHandler.py

```diff
@@ -2784,100 +2803,179 @@ def MetaDataTests(self):
-                raise e        
-            
-            # 3. Report Filesystem target space parameters for verification by user
-            Print("REPORT FILESYSTEM TARGET SPACE PARAMETERS FOR VERIFICATION BY THE USER")
+                Print("   - Functional testing failed with error: %s" % str(e))
+                retVal = False   
+
+            # Now perform some cleanup here
             try:
-                Print("  - %-20s: %s" % ('Total space', util.get_fs_size(testdir)))
-                Print("  - %-20s: %s" % ('Space utilization',util.get_fs_utilisation(testdir)))
-                displayOperationStatus(True)
+                if testFileCreated:
+                    os.remove(testfile)
+                if testDirCreated:
+                    os.rmdir(testdir)
+                if mountCreated:
+                    nfs.unmount(mountpoint, True)
                 checkPoints += 1
             except Exception, e:
-                Print("   - Failed to report filesystem space utilization parameters. " )
-                raise e 
-        except Exception, e:
-            Print("   - Functional testing failed with error: %s" % str(e))
-            retVal = False   
+                Print("   - Failed to cleanup after NFS functional tests, please delete the following manually: %s, %s, %s. Exception: %s" % (testfile, testdir, mountpoint, str(e)))
+
+        return (retVal, checkPoints, totalCheckPoints)
+
+    def ControlPathStressTests(self):
+        sr_ref = None 
+        retVal = True
+        checkPoint = 0
+        totalCheckPoints = 0
+        pbdPlugUnplugCount = 10
+
+        nfs_versions = self.getSupportedNFSVersions()
+        for nfsv in nfs_versions:
+            Print("Using NFSVersion: %s" % nfsv)
+            totalCheckPoints += 5
+            try:
+                Print("SR CREATION, PBD PLUG-UNPLUG AND SR DELETION TESTS")
+                Print(">> These tests verify the control path by creating an SR, unplugging")
+                Print("   and plugging the PBDs and destroying the SR in multiple iterations.")
+                Print("")
+            
+                for i in range(0, 10):
+                    Print("   -> Iteration number: %d" % i)
+                    totalCheckPoints += (2 + pbdPlugUnplugCount)
+                    (retVal, sr_ref, device_config) = self.Create(nfsv)
+                    if not retVal:                    
+                        raise Exception("      SR creation failed.")
+                    else:
+                        checkPoint += 1
+                
+                    # Plug and unplug the PBD over multiple iterations
+                    checkPoint += StorageHandlerUtil.PlugAndUnplugPBDs(self.session, sr_ref, pbdPlugUnplugCount)
+                    
+                    # destroy the SR
+                    Print("      Destroy the SR.")
+                    StorageHandlerUtil.DestroySR(self.session, sr_ref)
+                    checkPoint += 1
+                    
+
```

Comment:
We seem to have missed a block of prints here (It's in the original StorageHandler::ControlPathStressTests)

"SR SPACE AVAILABILITY TEST
>> This test verifies that all the free space advertised by an SR
   is available and writable."